tests/e2e/test_kb_builder_pipeline_e2e.py

```diff
@@ -0,0 +1,436 @@
+                                "end_node_id": "1"
+                            },
+                            {
+                                "type": "LED_BY",
+                                "start_node_id": "2",
+                                "end_node_id": "1"
+                            }
+                        ]
+                    }"""
+        ),
+        LLMResponse(content='{"nodes": [], "relationships": []}'),
+        LLMResponse(content='{"nodes": [], "relationships": []}'),
+    ]
+
+    # user input:
+    # the initial text
+    # and the list of entities and relations we are looking for
+    pipe_inputs = {
+        "splitter": {"text": text},
+        "schema": {
+            "entities": [
+                SchemaEntity(
+                    label="Person",
+                    properties=[
+                        SchemaProperty(name="name", type="STRING"),
+                        SchemaProperty(name="place_of_birth", type="STRING"),
+                        SchemaProperty(name="date_of_birth", type="DATE"),
+                    ],
+                ),
+                SchemaEntity(
+                    label="Organization",
+                    properties=[
+                        SchemaProperty(name="name", type="STRING"),
+                    ],
+                ),
+                SchemaEntity(
+                    label="Potion",
+                    properties=[
+                        SchemaProperty(name="name", type="STRING"),
+                    ],
+                ),
+                SchemaEntity(
+                    label="Location",
+                    properties=[
+                        SchemaProperty(name="address", type="STRING"),
+                    ],
+                ),
+            ],
+            "relations": [
+                SchemaRelation(
+                    label="KNOWS",
+                ),
+                SchemaRelation(
+                    label="PART_OF",
+                ),
+                SchemaRelation(
+                    label="LEAD_BY",
+                ),
+                SchemaRelation(
+                    label="DRINKS",
+                ),
+            ],
+            "potential_schema": [
+                ("Person", "KNOWS", "Person"),
+                ("Person", "DRINKS", "Potion"),
+                ("Person", "PART_OF", "Organization"),
+                ("Organization", "LEAD_BY", "Person"),
+            ],
+        },
+    }
+    res = await kg_builder_pipeline.run(pipe_inputs)
+    # llm must have been called for each chunk
+    assert llm.invoke.call_count == 3
+    # result must be success
+    assert res == {"writer": {"status": "SUCCESS"}}
+    # check component's results
+    chunks = kg_builder_pipeline.get_results_for_component("splitter")
+    assert len(chunks["chunks"]) == 3
+    graph = kg_builder_pipeline.get_results_for_component("extractor")
+    # 3 entities + 3 chunks
+    assert len(graph["nodes"]) == 6
```

Comment:
Is it worth asserting what the entities and relations are in this case? Instead of just asserting the number of them

Reply:
Done in https://github.com/neo4j/neo4j-genai-python/pull/98/commits/62aeb21fee174c9f240bc058400ea4ce6cb0de66 (and https://github.com/neo4j/neo4j-genai-python/pull/98/commits/32b31bc6f0a7f6159a66f5efa8c663cc6cb83a15 )